java/cdk/src/main/java/sleeper/cdk/stack/TransactionLogSnapshotStack.java

```diff
@@ -131,7 +145,74 @@ public TransactionLogSnapshotCreationStack(
         coreStacks.grantInvokeScheduled(snapshotCreationTrigger, queue);
         coreStacks.grantReadTablesStatus(snapshotCreationLambda);
         transactionLogStateStoreStack.grantCreateSnapshots(snapshotCreationLambda);
+    }
 
-        Utils.addStackTagIfSet(this, instanceProperties);
+    private void createSnapshotDeletionLambda(InstanceProperties instanceProperties, LambdaCode statestoreJar, CoreStacks coreStacks,
+            TransactionLogStateStoreStack transactionLogStateStoreStack, Topic topic, List<IMetric> errorMetrics) {
+        String triggerFunctionName = String.join("-", "sleeper",
+                instanceProperties.get(ID).toLowerCase(Locale.ROOT), "tl-snapshot-deletion-trigger");
+        String deletionFunctionName = String.join("-", "sleeper",
+                instanceProperties.get(ID).toLowerCase(Locale.ROOT), "tl-snapshot-deletion");
```

Comment:
This should work similarly to the changes in https://github.com/gchq/sleeper/pull/2594, with Utils.cleanInstanceId.